examples/server/server.cpp

```diff
@@ -2288,6 +2341,13 @@ struct server_context {
 
                         GGML_ASSERT(batch.n_tokens > 0);
 
+                        common_sampler_reset(slot.smpl);
+
+                        // Process all prompt tokens through sampler system
+                        for (int i = 0; i < slot.n_prompt_tokens; ++i) {
+                            common_sampler_accept(slot.smpl, prompt_tokens[i], false);
+                        }
+
```

Comment:
@ngxson @ggerganov Can I keep this code in the PR as a temporary bandaid since it seems to potentially function a bit better than the current state at present, and then after the PR merges this could be fixed more properly?

Reply:
@ngxson and @wwoodsTM This indeed looks like a bug and wasn't the intended way.

The changes here appear to be mostly correct, except the sampler will now accept the tokens even for embedding and reranking tasks, which is probably completely fine.

> Removing this will break the cached prompt. These tokens are not in prompt_tokens

I think the `prompt_tokens` should contain the full context, including the cached tokens, so this should be OK.

Like @ngxson mentioned earlier, I agree that the sampler changes in the server code would be better to be reviewed in a separate PR. @wwoodsTM Would you mind to open a new PR just for those changes alone - I think we can merge them fast and then you can rebase the DRY implementation on top of them?

Reply:
@ggerganov Makes total sense to me, thanks.

I just created #10019 for this, as requested.